hummingbot/connector/gateway/clob/gateway_sol_clob.py

```diff
@@ -0,0 +1,1252 @@
+        self._chain = chain
+        self._network = network
+        self._trading_pairs = trading_pairs
+        self._tokens = set()
+        [self._tokens.update(set(trading_pair.split("-"))) for trading_pair in trading_pairs]
+        self._wallet_address = wallet_address
+        self._trading_required = trading_required
+        self._ev_loop = asyncio.get_event_loop()
+        self._last_poll_timestamp = 0.0
+        self._last_balance_poll_timestamp = time.time()
+        self._last_est_gas_cost_reported = 0
+        self._allowances = {}
+        self._chain_info = {}
+        self._status_polling_task = None
+        self._get_chain_info_task = None
+        self._auto_approve_task = None
+        self._poll_notifier = None
+        self._native_currency = None
+        self._network_transaction_fee: Optional[TokenAmount] = TokenAmount('SOL', constant.FIVE_THOUSAND_LAMPORTS)
+        self._order_tracker: ClientOrderTracker = ClientOrderTracker(connector=self)
+        self._get_markets_task = None
+        self._markets = None
+        self._auto_create_token_accounts_task = None
+        self._tokens_accounts_created: bool = False
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        global s_logger
+        if s_logger is None:
+            s_logger = logging.getLogger(cls.__name__)
+
+        return cast(HummingbotLogger, s_logger)
+
+    @property
+    def chain(self):
+        return self._chain
+
+    @property
+    def network(self):
+        return self._network
+
+    @property
+    def connector(self):
+        return self._connector_name
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def address(self):
+        return self._wallet_address
+
+    # Added for compatibility
+    @property
+    def connector_name(self):
+        return self._connector_name
+
+    async def all_trading_pairs(self, chain: str, network: str) -> List[str]:
+        # Since the solana tokens trading pairs would be too much, we are returning an empty list here.
+        return []
+
+        # """
+        # Calls the token's endpoint on the Gateway.
+        # """
+        # try:
+        #     tokens = await self._get_gateway_instance().get_tokens(chain, network)
+        #     token_symbols = [token["symbol"] for token in tokens["tokens"]]
+        #     trading_pairs = []
+        #     for base, quote in it.permutations(token_symbols, 2):
+        #         trading_pairs.append(f"{base}-{quote}")
+        #
+        #     return trading_pairs
+        # except (Exception,):
+        #     GatewaySOLCLOB.logger().warning(f"""No trading pairs found for {chain}/{network}.""")
+        #
+        #     return []
+
+    @staticmethod
+    def is_order(in_flight_order: CLOBInFlightOrder) -> bool:
+        return in_flight_order.client_order_id.split("-")[0] in {"buy", "sell"}
+
+    # Added for compatibility
+    @staticmethod
+    def is_amm_order(in_flight_order: CLOBInFlightOrder) -> bool:
```

Comment:
Because the GatewaySOLCLOB class works the amm_arb strategy, and we adapted the implementation of the GatewayEVMAMM, we needed to use the same methods or properties used there. This was then added for compatibility since a more generic name would be "is_order" instead of "is_amm_order". But as you will be able to see, is_amm_order uses another method to fulfill the needed requirements.